core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -974,12 +974,12 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
       new SqlLikeOperator("NOT ILIKE", SqlKind.LIKE, true, false);
 
   /** The regex variant of the LIKE operator. */
-  @LibraryOperator(libraries = {SPARK, HIVE})
+  @LibraryOperator(libraries = {SPARK, HIVE, MYSQL})
   public static final SqlSpecialOperator RLIKE =
       new SqlLikeOperator("RLIKE", SqlKind.RLIKE, false, true);
 
   /** The regex variant of the NOT LIKE operator. */
-  @LibraryOperator(libraries = {SPARK, HIVE})
+  @LibraryOperator(libraries = {SPARK, HIVE, MYSQL})
```

Comment:
I think the `MySQL` engine should not be added. 
Although the SQL is the same, their results are not the same. Hive and Spark will return true/false, while MySQL will return 1/0

Reply:
@YiwenWu Thank you for your reminder. If we want to add RLIKE to MySQL, I think we should write an Operator, just like the log function, to adapt to different database dialects.

Reply:
@YiwenWu @caicancai In MySQL, Use 0 or 1 to replace the TRUE or false.
In MySQL Example:
`SELECT 1 = 1 FROM Product;`
will return 1 to replace the true.
And For example:
`SELECT * FROM Product where Name not rlike '.apper Plus';`
the NOT RLIKE can be used TRUE or False, So I think we don't add another operator for this.
Same as `REGEXP_LIKE` function in SqlLibraryOperators.

Reply:
Regarding the question of whether the return value needs to be completely consistent. I'm not sure, this is an ambiguous question. 
If Calcite allows the return value to be different from the corresponding engine, then it is okay to handle it this way.